umbra/umbra_64.c

```diff
@@ -1261,6 +1283,28 @@ umbra_insert_app_to_shadow_arch(void *drcontext,
                                 reg_id_t *scratch_regs,
                                 int num_scratch_regs)
 {
+#if defined(AARCH64)
+    reg_id_t tmp = *scratch_regs;
+    instru_insert_mov_pc(drcontext, ilist, where, opnd_create_reg(tmp),
+                                OPND_CREATE_INT64(map->disp));
+    PRE(ilist, where, INSTR_CREATE_add(drcontext,
```

Comment:
The instruction sequence is different from the existing one. One did you place the `and` instruction last here?